### ISSUE TITLE
Wrap Elixir ANSI implementation

### ISSUE THREAD

**Issue by ericmj** [hexpm/hex#541]
To make tests easier we have added `Application.put_env(:elixir, :ansi_enabled, false)` to test `test_helper.exs` but this disables all coloring and highlighting from `mix test`.

Wrap all ANSI functions so that we can use our own configuration that does not affect `mix test` output.

**Comment by supersimple** [comment]
Would this mean a new function that is defined conditionally based on mix env? 
```
if Mix.env() == :test do
  <function defined with wrapper to disable ANSI output>
else
  <function that delegates to IO.ANSI>
end
```

**Comment by ericmj** [comment]
We need our own config that determines if we should emit fragments or not, so something like this: https://github.com/elixir-lang/elixir/blob/91e8a76bfb5f064bd28b15e85f4ac5e84848b2b5/lib/elixir/lib/io/ansi.ex#L249 based on a `Hex.State` config.

**Comment by supersimple** [comment]
Resolved in https://github.com/hexpm/hex/pull/544

**Comment by ericmj** [comment]
Fixed in #544.
